### PR TITLE
Fixed unit tests

### PR DESCRIPTION
**Pull request by ghost**
Signed-off-by: Guian Gumpac <guiang@bitquilltech.com>

### Description
In order to fix unit tests the following needed to be modified:
- boost was removed as it is an obsolete feature
- interval was updated as it is obsolete and was replaced by to calendar_interval
- removed ml and LicensedExpired as it is an xpack feature
- removed labels as it is part of the implementation of the Flattened xpack feature
- removed IndicesStats, RecoveryStatus, and Segments as it is part of the monitoring xpack feature in the OSC client

### Additional Reviewers
@raymond-lum @alexmeizer @Yury-Fridlyand

### CODE REVIEW

**Review by alexmeizer — APPROVED** [review]

File: src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs [hunk 1]
```diff
@@ -113,7 +111,6 @@ internal static class ErrorCauseFormatterStatics
 			{ "index", 6 },
 			{ "index_uuid", 7 },
 			{ "lang", 8 },
-			{ "license.expired.feature", 9 },
```
Comment (raymond-lum) [inline src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs:116]: If we remove this line do we know if the number value needs to adjust downstream or is this important that the actual numbers stay as they are in the Dictionary? i.e. would changing the number 0, 1, 2,... 8, 10, break some compatibility?

File: src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs [hunk 2]
```diff
@@ -190,9 +187,6 @@ public TErrorCause Deserialize(ref JsonReader reader, IJsonFormatterResolver for
 								case 8:
 									errorCause.Language = reader.ReadString();
 									break;
-								case 9:
-									errorCause.LicensedExpiredFeature = reader.ReadString();
-									break;
```
Comment (raymond-lum) [inline src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs:195]: Similar question as we have now a break in the cases. I just don't know if it's OK but wanted to ensure this was checked.

File: tests/Tests/Aggregations/Pipeline/SumBucket/SumBucketAggregationUsageTests.cs [hunk 3]
```diff
@@ -46,7 +46,7 @@ public class SumBucketAggregationUsageTests : AggregationUsageTestBase<ReadOnlyC
 				date_histogram = new
 				{
 					field = "startedOn",
-					interval = "month",
+					calendar_interval = "month",
```
Comment (raymond-lum) [inline tests/Tests/Aggregations/Pipeline/SumBucket/SumBucketAggregationUsageTests.cs:49]: Why was this changed in the structure?

Comment (ghost) [reply tests/Tests/Aggregations/Pipeline/SumBucket/SumBucketAggregationUsageTests.cs:49]: interval was marked as obsolete and was therefore removed. Reference: https://github.com/elastic/elasticsearch-net/blob/0b2a83b8f9647ae482c91e67fe9bf983d18c2947/src/Nest/Aggregations/Bucket/DateHistogram/DateHistogramAggregation.cs#L45

File: src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs (hunk 1 above)
Comment (ghost) [reply src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs:116]: I will test this out. Thanks

Comment (ghost) [reply src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs:116]: Updated the changes to adjust the numbers downstream.

File: src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs (hunk 2 above)
Comment (ghost) [reply src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs:195]: Updated the changes to adjust the numbers downstream.

File: src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs (hunk 1 above)
Comment (Yury-Fridlyand) [reply src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs:116]: Are this numbers or strings received from the server? Are they shifted in `opensearch` as well?

**Review by raymond-lum — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

File: src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs (hunk 1 above)
Comment (ghost) [reply src/OpenSearch.Net/Responses/ServerException/ErrorCause.cs:116]: Based on https://github.com/Bit-Quill/opensearch-net/blob/main/tests/Tests/ClientConcepts/ServerError/ComplexErrorTests.cs, the number values are not used.
